### PR TITLE
[java_17_migration] Migrate build to Java-17

### PR DESCRIPTION
**Pull request by gabilang**
## Purpose
$title
## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,8 +1,8 @@
 org.gradle.caching=true
 group=io.ballerina.stdlib
-version=2.3.0-SNAPSHOT
+version=3.0.0-SNAPSHOT
```
Comment (warunalakshitha) [inline gradle.properties:3]: Do we need to change version?

File: ballerina/Dependencies.toml [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 [ballerina]
 dependencies-toml-version = "2"
-distribution-version = "2201.7.0-20230619-175900-bb4e4544"
+distribution-version = "2201.7.0-SNAPSHOT"
```
Comment (warunalakshitha) [inline ballerina/Dependencies.toml:8]: We do not need this change

File: gradle.properties (hunk 1 above)
Comment (gabilang) [reply gradle.properties:3]: Probably a major version needed to be released with Java 17 support. For now I'll restore the changes.

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**